File: Swift/백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.swift
import Foundation

let heightArray = stride(from:0, to:9, by:1)
.map { _ in Int(readLine()!)! }
.sorted(by:<)

var queue: [(totalHeight: Int, heightArray: [Int])] = [(heightArray.reduce(0, +), heightArray)]

var result = [Int]()

var isEnd = false

while !isEnd {
    let element = queue.removeFirst()
    for i in element.heightArray.indices {
        var heightArray = element.heightArray
        let num = heightArray[i]
        let totalHeight = element.totalHeight - num
        heightArray.remove(at: i)
        if heightArray.count == 7 && totalHeight == 100 {
            result = heightArray
            isEnd.toggle()
        } else if heightArray.count != 7 {
            queue.append((totalHeight, heightArray))
        }
    }
}

result.forEach { print($0) }

File: Python/백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.py
from collections import namedtuple
import copy

height_list = sorted([int(input()) for _ in range(9)])

nt = namedtuple('nt', ['height_list', 'total_height'])

queue = [nt(height_list, sum(height_list))]

result = []

isEnd = False

while not isEnd:
    element = queue.pop(0)
    len_list = len(element.height_list)
    
    for i in range(len_list):
        temp_list = copy.deepcopy(element.height_list)
        num = temp_list[i]
        total_height = element.total_height - num
        temp_list.pop(i)
        if len_list != 8:
            queue.append(nt(temp_list, total_height))
        elif len_list == 8 and total_height == 100:
            isEnd = True
            result = temp_list
                
for num in result:
    print(num)